EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/GeneratePanel.java

```diff
@@ -146,34 +149,51 @@ protected Object load() {
-                }
-                attributeInput.put(attribute, "");
+                final AttributeInput attributeInput = AttributeInput.attributeInputFactory(attribute);
+                item.add(new Label("attribute", attributeInput.getAttributeName()));
+                item.add(new Label("attributeType", attributeInput.getAttributeType().getName()));
+                final Label attributeInputDescriptionLabel = new Label("attributeInputDescription", getAttributeInputDescription(attributeInput));
+                attributeInputDescriptionLabel.setOutputMarkupId(true);
+                item.add(attributeInputDescriptionLabel);
+                attributeInputs.add(attributeInput);
                 IModel<String> attributeInputModel = new Model<String>() {
                     @Override
                     public String getObject() {
-                        return attributeInput.get(attribute);
+                        int indexOfAttributeInput = attributeInputs.indexOf(attributeInput);
+                        return attributeInputs.get(indexOfAttributeInput).getInput();
                     }
                     @Override
                     public void setObject(String inputValue) {
-                        attributeInput.put(attribute, inputValue);
+                        int indexOfAttributeInput = attributeInputs.indexOf(attributeInput);
+                        attributeInputs.get(indexOfAttributeInput).setInput(inputValue);
                     }
                 };
 
-                TextField<String> inputField = new TextField<>("attributeInput", attributeInputModel);
-                inputField.add(AttributeValidator.getValidatorForAttribute(attribute));
+                final TextField<String> inputField = new TextField<>("attributeInput", attributeInputModel);
+                inputField.add(attributeInput.getAttributeInputValidator());
                 inputField.setLabel(new Model<String>(attribute.getName()));
+                inputField.setOutputMarkupId(true);
                 item.add(inputField);
-                item.add(new Label("attributeInputWarning", "")
-                        .setVisible(attributeDependencyManager.isDependentAttributeInAnyDependency(attribute)));
+                Boolean isDependentAttribute = attributeDependencyManager.isDependentAttributeInAnyDependency(attribute);
+                item.add(new Label("attributeInputWarning", "").setVisible(isDependentAttribute));
+
+                DropDownChoice<AttributeInput.ProbabilityDistributionEnum> methodDropDown = new DropDownChoice<>(
+                        "attributeInputMethodSelection",
+                        new PropertyModel<AttributeInput.ProbabilityDistributionEnum>(attributeInput, "selectedMethod"),
+                        attributeInput.getAvailableMethods());
+                methodDropDown.add(new AjaxFormComponentUpdatingBehavior("onChange") {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This class has 6 parents which is greater than 5 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)